inc/features/blocks/personalization/components/variant-analytics.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Views from './views';
+
+const { useSelect } = wp.data;
+
+const VariantAnalytics = ( { variant } ) => {
+	const { audience, fallback } = variant.attributes;
+
+	// Show nothing if no audience selected and not the fallback.
+	if ( ! fallback && ! audience ) {
+		return null;
+	}
+
+	const postId = useSelect( select => {
+		return select( 'core/editor' ).getCurrentPostId();
+	} );
+
+	// No post ID so post isn't published, don't show anything.
+	if ( ! postId ) {
+		return null;
+	}
+
+	// Get the XB variant parent client ID.
+	const clientId = useSelect( select => {
+		const { getBlockAttributes, getBlockRootClientId } = select( 'core/block-editor' );
+		const parentClientId = getBlockRootClientId( variant.clientId );
+		return getBlockAttributes( parentClientId ).clientId;
+	}, [ variant.clientId ] );
+
+	// Fetch the stats.
+	const views = useSelect( select => {
+		return select( 'analytics/xbs' ).getViews( clientId, postId );
+	}, [ clientId, postId ] );
+	const isLoading = useSelect( select => {
+		return select( 'analytics/xbs' ).getIsLoading();
+	}, [ views ] );
+
+	const audienceId = audience || 0;
+
+	const audienceViews = views?.audiences?.
+		find( audienceData => audienceData.id === audienceId )?.
+		count || 0;
```

Comment:
I find this pretty unreadable; can we either break out into multiple vars, or condense to one line?